mwdb/core/search/fields.py

```diff
@@ -152,6 +152,32 @@ def get_condition(self, expression: Expression, remainder: List[str]) -> Any:
             return self.column.any(self.value_column == value)
 
 
+class FavoritesField(BaseField):
+    def get_condition(self, expression: Expression, remainder: List[str]) -> Any:
+        if remainder:
+            raise FieldNotQueryableException(
+                f"Field doesn't have subfields: {'.'.join(remainder)}"
+            )
+
+        value = get_term_value(expression)
+
+        if not g.auth_user.has_rights(Capabilities.manage_users) and g.auth_user.login != value:
+            raise ObjectNotFoundException("Only the mwdb admin can search for other users favorites")
+
+        if g.auth_user.login == value:
+            user = g.auth_user
+        else:
+            user = db.session.query(User).filter(User.login == value).first()
+
+        if user is None:
+            raise ObjectNotFoundException(f"No such user: {value}")
+
+        favorites_id = [favorite.id for favorite in user.favorites]
+        return self.column.any(
+            ObjectPermission.related_object_id.in_(favorites_id)
+        )
```

Comment:
It shouldn't be related with shares. Favorites are separate list.
```suggestion
        return self.column.any(User.login == user)
```
where `self.column` is `Object.followers`. See the similar implementation for ListField and tags.